From oliver@neukum.org  Mon Sep 15 14:03:13 2008
From: Oliver Neukum <oliver@neukum.org>
Date: Mon, 15 Sep 2008 17:29:28 +0200
Subject: USB: RESET_RESUME needs to block autosuspend when remote wakeup is needed
To: Alan Stern <stern@rowland.harvard.edu>
Cc: linux-usb@vger.kernel.org, greg@kroah.com
Message-ID: <200809151729.28639.oliver@neukum.org>
Content-Disposition: inline


Reset upon resumption will wipe the input buffer and is therefore
a reason to not suspend if remote wakeup is requested because
the driver needs that data.

Signed-off-by: Oliver Neukum <oneukum@suse.de>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/core/driver.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/drivers/usb/core/driver.c
+++ b/drivers/usb/core/driver.c
@@ -1070,7 +1070,8 @@ static int autosuspend_check(struct usb_
 				struct usb_driver *driver;
 
 				driver = to_usb_driver(intf->dev.driver);
-				if (!driver->reset_resume)
+				if (!driver->reset_resume ||
+				    intf->needs_remote_wakeup)
 					return -EOPNOTSUPP;
 			}
 		}
